### PR TITLE
Add jump instruction for vm

### PR DESCRIPTION
**Pull request by RenfengLiu**

### CODE REVIEW

File: gapir/cc/interpreter.cpp [hunk 1]
```diff
@@ -469,6 +469,22 @@ Interpreter::Result Interpreter::switchThread(uint32_t opcode) {
   return CHANGE_THREAD;
 }
 
+Interpreter::Result Interpreter::addJumpLabel(uint32_t opcode) {
+  auto jump_id = extract26bitData(opcode);
+  mJumpLables[jump_id] = mCurrentInstruction;
+  return SUCCESS;
```
Comment (ben-clayton) [inline gapir/cc/interpreter.cpp]: `return mStack.isValid() ? SUCCESS : ERROR;`

File: gapir/cc/interpreter.cpp [hunk 2]
```diff
@@ -469,6 +469,22 @@ Interpreter::Result Interpreter::switchThread(uint32_t opcode) {
...
+}
+
+Interpreter::Result Interpreter::jumpNZ(uint32_t opcode) {
+  auto jump_id = extract26bitData(opcode);
+  auto should_jump = mStack.pop<int32_t>();
+  if (should_jump != 0) {
+    mCurrentInstruction = mJumpLables[jump_id];
+  }
+
+  return SUCCESS;
```
Comment (ben-clayton) [inline gapir/cc/interpreter.cpp]: `return mStack.isValid() ? SUCCESS : ERROR;`

File: gapir/cc/interpreter.cpp [hunk 3]
```diff
@@ -469,6 +469,22 @@ Interpreter::Result Interpreter::switchThread(uint32_t opcode) {
...
+}
+
+Interpreter::Result Interpreter::jumpNZ(uint32_t opcode) {
+  auto jump_id = extract26bitData(opcode);
+  auto should_jump = mStack.pop<int32_t>();
+  if (should_jump != 0) {
+    mCurrentInstruction = mJumpLables[jump_id];
+  }
+
```
Comment (ben-clayton) [inline gapir/cc/interpreter.cpp]: Should we assert the stack is zero length here? Carrying across stack values seems like a really bad idea.

File: gapir/replay_service/vm.h [hunk 4]
```diff
@@ -42,6 +42,8 @@ enum class Opcode {
   ADD = 14,
   LABEL = 15,
   SWITCH_THREAD = 16,
+  ADD_JUMP_LABEL = 17,
```
Comment (ben-clayton) [inline gapir/replay_service/vm.h]: The `ADD_` prefix seems a bit superfluous. Maybe just `JUMP_LABEL`?

File: gapis/replay/builder/builder.go [hunk 5]
```diff
@@ -277,9 +277,12 @@ func (b *Builder) CommitCommand() {
 				b.instructions[s.idx] = i
 				pop--
 			}
-		case asm.Clone, asm.Push, asm.Load: // Remove unused clones, pushes, loads
+		// case asm.Clone, asm.Push, asm.Load: // Remove unused clones, pushes, loads
```
Comment (ben-clayton) [inline gapis/replay/builder/builder.go]: Remove commented out code.

File: gapis/replay/builder/builder.go [hunk 6]
```diff
@@ -277,7 +277,8 @@ func (b *Builder) CommitCommand() {
...
```
Comment (AWoloszyn) [inline gapis/replay/builder/builder.go]: We shouldn't have commented out code here.

**Pull request merged**